File: haskell/adventofcode2015/src/Day01.hs
module Day01 (main, part1, part2) where

main :: IO ()
main = interact (show . \input -> (part1 input, part2 input))

parse :: String -> [Int]
parse = map (\p -> if p == '(' then 1 else -1)

part1 :: String -> Int
part1 = sum . parse

part2 :: String -> Int
part2 = length . takeWhile (>= 0) . scanl (+) 0 . parse

File: adventofcode-python/aoc2022/src/Day01.py
from aocd import data, submit


def parse(input):
    return [sum([int(y) for y in x.splitlines()]) for x in input.strip().split("\n\n")]


def part1(input):
    return max(parse(input))


def part2(input):
    return sum(sorted(parse(input), reverse=True)[0:3])


parts = (part1, part2)


if __name__ == "__main__":
    part1Answer = part1(data)
    submit(part1Answer, part="a")
    print("part 1: %s" % part1Answer)

    part2Answer = part2(data)
    submit(part2Answer, part="b")
    print("part 2: %s" % part2Answer)
